cloud-nio/cloud-nio-impl-ftp/src/test/scala/cloud/nio/impl/ftp/FtpCloudNioFileSystemProviderSpec.scala

```diff
@@ -42,30 +43,33 @@ class FtpCloudNioFileSystemProviderSpec extends AnyFlatSpec with CromwellTimeout
   }
 
   it should "pre compute the size before opening a read channel to avoid deadlocks" in {
-    val mockSizeFunction = mockFunction[Long]
-    val provider = new FtpCloudNioFileSystemProvider(ConfigFactory.empty(), FtpAnonymousCredentials, ftpFileSystems) {
+    val mockSizeFunction = mock[() => Long]
+    val provider: FtpCloudNioFileSystemProvider = new FtpCloudNioFileSystemProvider(
+      ConfigFactory.empty, FtpAnonymousCredentials, ftpFileSystems
+    ) {
 
-      override def fileProvider = new FtpCloudNioFileProvider(this) {
-        override def fileAttributes(cloudHost: String, cloudPath: String) =
+      override def fileProvider: FtpCloudNioFileProvider = new FtpCloudNioFileProvider(this) {
+        override def fileAttributes(cloudHost: String, cloudPath: String): Option[CloudNioRegularFileAttributes] =
           Option(
             new CloudNioRegularFileAttributes {
               override def fileHash = throw new UnsupportedOperationException()
               override def lastModifiedTime() = throw new UnsupportedOperationException()
-              override def size() = mockSizeFunction()
+              override def size(): Long = mockSizeFunction()
               override def fileKey() = throw new UnsupportedOperationException()
             }
           )
 
-        override def read(cloudHost: String, cloudPath: String, offset: Long) = {
+        override def read(cloudHost: String, cloudPath: String, offset: Long): ReadableByteChannel = {
           mock[ReadableByteChannel]
         }
       }
     }
 
     // This should only be called once, not every time we ask for the channel size
-    mockSizeFunction.expects().onCall(_ => 60).once()
+    when(mockSizeFunction.apply()).thenReturn(60)
```

Comment:
Are the semantics of `thenReturn` the same as the `once()`?

Reply:
Unfortunately, the new implicits are NOT as fancy scalamock. scalamock makes sophisticated wrappers that allow DSLs like `.once()`. At least in Scala 2.x 😉

Instead down below there's a call to:

```scala
verify(mockSizeFunction).apply()
```

That is an equivalent verification to `.once()` ensuring that `.apply()` was called once and only once on the mock.

For the lurkers, the idea behind bypassing scalamock and using the Mockito Java API directly is further discussed in the scaladocs for `trait MockSugar`.